### ISSUE TITLE
Fix translation process for plurals, sentence combinations, ...

### ISSUE THREAD

**Issue by Boldewyn** [Codepoints/Codepoints.net#84]
https://twitter.com/nerdblogpl/status/712555068978028544

>  then maybe your translation process is broken because this doesn't sound like a correct sentence:

I assume the problem is, that I really do split strings at unfortunate places for some use cases. We should look into [MessageFormat](http://php.net/manual/en/class.messageformatter.php) more closely.

**Comment by d4rky-pl** [comment]
Examples of broken translations:

> Teoretycznie powinne to być wszystkie znaki kiedykolwiek używane. 

should be

> Teoretycznie powinny to być wszystkie kiedykolwiek używane znaki

> W praktyce Unicode zawiera 120803 współrzędnych kodowych aktualnie zdefiniowanych, odwzorowujących znaki od egipskich hieroglifów do ornamentów i symboli.

should be

> W praktyce Unicode zawiera 120803 aktualnie zdefiniowanych współrzędnych kodowych, odwzorowujących znaki od egipskich hieroglifów do ornamentów i symboli.

> Przeglądać je pojedynczo zaczynając od pierwszej współrzędnej, ␀ lub szukaj konkretnego znaku. 

should be

> Przeglądaj je pojedynczo, zaczynając od pierwszej współrzędnej, ␀ lub szukaj konkretnego znaku. 

From footer:

> Zawartość może być swobodnie używane ponownie zgodnie z podanymi warunkami.

should be

> Zawartość może być swobodnie używana ponownie zgodnie z podanymi warunkami.

There are also several typos or sentences that sound unnatural but that's just the quality of the translation and not the issue itself.

**Comment by d4rky-pl** [comment]
From the look of your `.po` files it seems that most of the issues are actually because of typos / weird sentence order in translations themselves, not something that needs to be fixed in code. I will look into it when I get some time.

**Comment by Boldewyn** [comment]
Thanks for the feedback! If you want to work on the translation, it'd be cool, if you'd hop over to crowdin.com/project/codepoints and not edit the `.po` files directly here. That way the translations can be discussed and reviewed better.

@jsbien: Are you OK with this? Any opinions?

Independently, I _will_ have to work on the codebase no matter what. I was already quite sure, that the current mess will break for languages like Chinese, Arabic or Hebrew. Biggest obstacle is correctly pluralizing a sentence with more than one variable. Here, gettext is utterly and famously useless.

Hence me looking into ICU's MessageFormat. However, I'll keep all you translators posted, before I introduce any breaking changes.

**Comment by jsbien** [comment]
I'm afraid that from a long term point of view some changes on the code level would be desirable. I found the messages difficult to translate because they are usually very short and the context was not easily accessible. So actually the best way to check is to see the full translation on the site. I intend to do it locally before submitting a new translation, but don't have time for setting the system now. 

On the other hand, I remember vaguely that you can put in the source some comments for translators.

I think a lot of cases of using the correct plural can be handled by gettext. I have no opinion about ICU Message format. I don't remember if/how XLIFF is handling plurals with two variables. At the moment the plurals have to be quite often wrong.

Some errors pointed by Michał Matyas consist in wrong gender agreement. I have to check what can be done about. If a phrase is used only once, then the mistake is due to the lack of context information and can be corrected without problem. If the phrase is re-used several times, then it is more difficult.

To summarize: I also thank Michał Matyas for feedback and invite him to cooperate on Crowdin. In a view days I hope to correct at least some errors.

**Comment by jsbien** [comment]
I've updated Crowdin with Michał's correction. Thanks again for the feedback. I will review the whole translation after Easter.

**Comment by d4rky-pl** [comment]
@jsbien glad I could help! Ping me here after you review and update it and I'll help you proofread it again :)

**Comment by jsbien** [comment]
Please use my corrected translation. I hope this time there are no technical problems.

**Comment by Boldewyn** [comment]
The translation is updated and live. Thanks again for the effort!
